src/internal/packager/images/pull.go

```diff
@@ -308,6 +311,23 @@ func (i *ImageConfig) PullImage(src string, spinner *message.Spinner) (img v1.Im
 			return nil, false, fmt.Errorf("failed to load image from docker daemon: %w", err)
 		}
 	} else {
+		if strings.Contains(src, "@") && (desc.MediaType == ctypes.OCIImageIndex || desc.MediaType == ctypes.DockerManifestList) {
+			imageOptions := "please select one of the images below based on your platform to use instead"
+			imageBaseName := strings.Split(src, "@")[0]
+			manifest, err := crane.Manifest(src, config.GetCraneOptions(i.Insecure)...)
+			if err != nil {
+				return nil, false, fmt.Errorf("%w: %w", lang.ErrUnsupportedImageType, err)
+			}
+			var idx v1.IndexManifest
+			if err := json.Unmarshal(manifest, &idx); err != nil {
+				return nil, false, fmt.Errorf("%w: %w", lang.ErrUnsupportedImageType, err)
+			}
+			for _, manifest := range idx.Manifests {
+				imageOptions = fmt.Sprintf("%s\n %s@%s for platform %s", imageOptions, imageBaseName, manifest.Digest, manifest.Platform)
+			}
+			return nil, false, fmt.Errorf("%w: %s", lang.ErrUnsupportedImageType, imageOptions)
+		}
```

Comment:
A couple comments on this block of code:

  - I think it would be more clean and less error prone to use a library to parse the image reference
  - we should be able to use the `desc.Manifest` directly rather than fetch the manifest from remote again
  
     Example:

    ```go
    reference, err := name.ParseReference(src)
    if err != nil {
	    return nil, false, err
    }
    
    if _, ok := reference.(name.Digest); ok && (desc.MediaType == ctypes.OCIImageIndex || desc.MediaType == ctypes.DockerManifestList) {
	      var idx v1.IndexManifest
	      if err := json.Unmarshal(desc.Manifest, &idx); err != nil {
	          return nil, false, fmt.Errorf("%w: %w", lang.ErrUnsupportedImageType, err)
	      }
	      
	      imageBaseName := path.Join(reference.Context().Registry.RegistryStr(), reference.Context().RepositoryStr())
    
    }
    ```
    
    if we went this route, we would have to update the test to not require `stderr` to contain an image reference with the tag in it because images that contain a tag + sha are technically considered a digest rather than a tag, so the tag is essentially ignored